scripts/get_objf_and_derivs_split.py

```diff
@@ -351,6 +357,57 @@ def MergeAndComputeObjfForSplit(split_index):
     ComputeObjfAndFinalDerivs(split_index, args.derivs_out != None)
 
+def CleanUpIfNeeded():
+    # For cleaning up intermediate files that won't be used, note if option need-model = true
+    # be used, clean up will keep file 'float.all'
+    if args.clean_up == 'true':
+        try:
+            if not os.path.isdir(args.work_dir):
+                ExitProgram("error finding working directory {0}".format(args.work_dir))
+            # If need_split_float option set to be true, this script will keep "float.all" in
+            # each split dir for future use
+            need_split_float = True if args.need_split_float == 'true' else False
+            for split_index in range(1, args.num_splits + 1):
+                split_dir = '{0}/split{1}/{2}'.format(args.work_dir,args.num_splits,split_index)
+                if not os.path.isdir(split_dir):
+                    ExitProgram("error finding split directory {0}".format(split_dir))
+                for file in os.listdir(split_dir):
+                    if re.match('(discounted|discounted_derivs|float|float_derivs|merged|merged_derivs)\.',file):
+
+                        if need_split_float and file == "float.all":
+                            continue
+                        os.remove(split_dir+"/"+file)
+                if need_split_float: continue
+                if os.listdir(split_dir) == []:
+                    os.rmdir(split_dir)
+                else:
+                    print("get_objf_and_drivs_split.py: split dir {0} in working dir has unexpected file,"\
+                          " check it if nesscessary".format(split_dir),file=sys.stderr)
+            if not need_split_float:
+                all_split_dir = '{0}/split{1}'.format(args.work_dir,args.num_splits)
+                if os.listdir(all_split_dir) == []:
+                    os.rmdir(all_split_dir)
+                else:
+                    print("get_objf_and_drivs_split.py: split dir {0} in working dir has unexpected file,"\
+                          " check it if nesscessary".format(all_split_dir),file=sys.stderr)
+            # If need_model option set to be true, this script will keep "float.all" in working dir
+            need_float = True if args.need_model == 'true' else False
+            for file in os.listdir(args.work_dir):
+                    if re.match('(discounted|discounted_derivs|float|float_derivs|merged|merged_derivs)\.',file):
+                        if need_float and file == "float.all":
+                            continue
+                        os.remove(args.work_dir+"/"+file)
+        except:
+            sys.exit("get_objf_and_drivs_split.py: error removing files in working dir")
+    try:
+        f = open(args.work_dir+'/cleaned', "w")
+        print(args.clean_up, file=f)
+        f.close()
+    except:
+        sys.exit("get_objf_and_derivs_split.py: error opening --cleaned={0} for writing".format(
+                  args.work_dir+'/cleaned'))
+
+
```

Comment:
FYI: for the top-level directory, I didn't want it to remove any files that have names not equal to those made by the script.  People might put other stuff there for some reason and not expect them to be removed.
What I was really hoping was that they could have the script just remove individually by name all the files it created, rather than messing about either with regexp's or searching through directories. This is safer, and not really that hard to implement.